packaging/language/pip.py

```
@@ -288,11 +288,17 @@ def _get_pip(module, env=None, executable=None):
             opt_dirs = []
         else:
             # Try pip with the virtualenv directory first.
-            opt_dirs = ['%s/bin' % env]
-        for basename in candidate_pip_basenames:
-            pip = module.get_bin_path(basename, False, opt_dirs)
-            if pip is not None:
-                break
+            opt_dirs = [os.path.join(env,'bin')]
+            for basename in candidate_pip_basenames:
+                pip = module.get_bin_path(basename, False, opt_dirs)
+                if pip is not None and pip == os.path.join(env,'bin',basename):
+                    break
+
+        if pip is None:
+          for basename in candidate_pip_basenames:
+              pip = module.get_bin_path(basename, False, opt_dirs)
+              if pip is not None:
+                  break
```

This looks like it might not go far enough.  I think if we're operating on a virtualenv we should check only the bin paths in the virtualenv.  If pip does not exist there we should return an error.